### PR TITLE
[Localnet] deduplicate service creation

### PR DESCRIPTION
**Pull request by SaveTheRbtz**
While here:
* added labels to observer logs
* added network, role, and num attributes to the trace emitter to be consistent w/ logs and metrics via `OTEL_RESOURCE_ATTRIBUTES`.

### CODE REVIEW

File: integration/localnet/bootstrap.go [hunk 1]
```diff
@@ -317,41 +317,25 @@ func prepareServiceDirs(role string, nodeId string) (string, string) {
...
+		"--datadir=/data/protocol",
+		"--secretsdir=/data/secret",
+		"--loglevel=DEBUG",
+		fmt.Sprintf("--profiler-enabled=%t", profiler),
+		fmt.Sprintf("--tracer-enabled=%t", tracing),
+		"--profiler-dir=/profiler",
+		"--profiler-interval=2m",
+		fmt.Sprintf("--admin-addr=:%v", AdminToolPort),
+	}
+	service.Volumes = []string{
```
Comment (pattyshack) [inline integration/localnet/bootstrap.go:333]: also dedup this while you're at it?

File: integration/localnet/bootstrap.go [hunk 2]
```diff
@@ -317,41 +317,25 @@ func prepareServiceDirs(role string, nodeId string) (string, string) {
...
-			"BINSTAT_DMP_PATH",
-		},
-		Labels: map[string]string{
-			"com.dapperlabs.role": container.Role.String(),
-			"com.dapperlabs.num":  fmt.Sprintf("%03d", i+1),
-		},
+	service := defaultService(container.Role.String(), i)
+	service.Command = []string{
+		fmt.Sprintf("--nodeid=%s", container.NodeID),
+		"--bootstrapdir=/bootstrap",
```
Comment (pattyshack) [inline integration/localnet/bootstrap.go:323]: a bunch of flags are also shared here.  maybe pre populate those as well?

File: integration/localnet/bootstrap.go [hunk 3]
```diff
@@ -503,37 +474,12 @@ func prepareObserverService(i int, observerName string, agPublicKey string) Serv
...
-		fmt.Sprintf("--profiler-enabled=%t", profiler),
-		fmt.Sprintf("--tracer-enabled=%t", tracing),
-		"--profiler-dir=/profiler",
-		"--profiler-interval=2m",
-	}
-	observerService.Volumes = []string{
-		fmt.Sprintf("%s:/bootstrap:z", BootstrapDir),
+	}...)
+	observerService.Volumes = append(observerService.Volumes, []string{
 		fmt.Sprintf("%s:/profiler:z", profilerDir),
```
Comment (pattyshack) [inline integration/localnet/bootstrap.go:479]: i feel dumb.  why can't these be dedup?  just pass in extra arguments to the defaultService function ....

Comment (SaveTheRbtz) [reply integration/localnet/bootstrap.go]: it's not you being dumb -- it's me being lazy =). Thanks for raising the bar!

File: integration/localnet/bootstrap.go [hunk 4]
```diff
@@ -317,64 +317,15 @@ func prepareServiceDirs(role string, nodeId string) (string, string) {
...
-			"BINSTAT_DMP_NAME",
-			"BINSTAT_DMP_PATH",
-		},
-		Labels: map[string]string{
-			"com.dapperlabs.role": container.Role.String(),
-			"com.dapperlabs.num":  fmt.Sprintf("%03d", i+1),
-		},
-	}
+	service := defaultService(container.Role.String(), dataDir, profilerDir, i)
+	service.Command = append(service.Command, []string{
```
Comment (pattyshack) [inline integration/localnet/bootstrap.go:321]: skip creating the slice, you're appending a single item here

**Review by pattyshack — APPROVED** [review]

File: integration/localnet/bootstrap.go (hunk 4 above)
Comment (SaveTheRbtz) [reply integration/localnet/bootstrap.go:321]: that one is left for the convenience of the next person who would want to add a new arg here.

Comment (pattyshack) [reply integration/localnet/bootstrap.go:321]: I still don't see the point of creating the slice first, then "...".  append is a vararg built-in function

https://go.dev/play/p/aoSfOM6bxkU

File: integration/localnet/bootstrap.go [hunk 5]
```diff
@@ -508,18 +459,37 @@ func prepareObserverService(i int, observerName string, agPublicKey string) Serv
 	// Observers have a unique naming scheme omitting node id being on the public network
 	dataDir, profilerDir := prepareServiceDirs(observerName, "")
 
-	observerService := Service{
-		Image: fmt.Sprintf("localnet-%s", DefaultObserverName),
+	observerService := defaultService(DefaultObserverName, dataDir, profilerDir, i)
+	observerService.Command = append(observerService.Command, []string{
```
Comment (pattyshack) [inline integration/localnet/bootstrap.go:463]: same here.  just get rid of `[]string{}...` altogether and directly append to command

**Review by peterargue — APPROVED** [review]
thanks for cleaning this up!

**Comment by SaveTheRbtz** [comment]
bors merge

**Pull request merged**